database/sql/src/main/scala/cromwell/database/slick/WorkflowStoreSlickDatabase.scala

```diff
@@ -62,6 +62,32 @@ trait WorkflowStoreSlickDatabase extends WorkflowStoreSqlDatabase {
                                      workflowStateTo: String,
                                      workflowStateExcluded: String)
                                     (implicit ec: ExecutionContext): Future[Seq[WorkflowStoreEntry]] = {
+
+    def updateForFetched(cromwellId: String,
+                         heartbeatTimestampTo: Timestamp,
+                         workflowStateFrom: String,
+                         workflowStateTo: String)
+                        (workflowStoreEntry: WorkflowStoreEntry)
+                        (implicit ec: ExecutionContext): DBIO[Unit] = {
+      val workflowExecutionUuid = workflowStoreEntry.workflowExecutionUuid
+      val updateState = workflowStoreEntry.workflowState match {
+        case matched if matched == workflowStateFrom => workflowStateTo
+        case other => other
+      }
+
+      for {
+        // When fetched, the heartbeat timestamp is set to now so we don't pick it up next time.
+        updateCount <- dataAccess
+          .workflowStoreFieldsForPickup(workflowExecutionUuid)
+          .update((updateState, Option(cromwellId), Option(heartbeatTimestampTo)))
+        _ <- assertUpdateCount(
+          s"Update $workflowExecutionUuid to $updateState, heartbeat timestamp to $heartbeatTimestampTo",
+          updateCount,
+          1
+        )
+      } yield ()
+    }
```

Comment:
No real changes here. This was really just a helper function in disguise before and is now scoped as one.